### PR TITLE
Implemented tab-conditions

### LINKED ISSUE #1870: Excerpt tab doesn't show on internal and external link content type
Currently the excerpt tab is not shown if the selected content type of a page internal or external link, like shown in the screenshot:

image

This might also be useful for external links, so that they can be assigned a tag or category and e.g. also be shown on the startpage using the smart content.

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #1870
| Related issues/PRs |  https://github.com/massiveart/husky/pull/614
| License | MIT
| Documentation PR | https://github.com/sulu-io/sulu-docs/pull/186

#### What's in this PR?

This PR add display conditions to hide/show tabs with conditions on the content on the record.

#### Why?

If you want only to show a tab in case of a special template in content-section.

#### Example Usage

__ContentNavigationProvider.php:__
~~~php
$test = new ContentNavigationItem('test');
$test->setId('tab-test');
$test->setAction('test');
$test->setDisplay(['edit']);
$test->setDisplayConditions(
    [
        new DisplayCondition('template', DisplayCondition::OPERATOR_EQUAL, 'default'),
    ]
);
~~~

__content/main.js:__
~~~javascript
this.sandbox.on('sulu.content.contents.saved', function(id, data) {
    this.sandbox.emit('husky.header.saved', data);
});

header: function(){
     return {
         tabs: {
             url: navigationUrl,
             componentOptions: {
                 conditionData: this.content.toJSON()
             }
        }
    };
}
~~~

#### To Do

- [x] Create a documentation PR
- [x] Introduce `saved` event
- [x] Introduce this in other bundles
- [x] roles
- [x] snippets

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Admin/ContentContentNavigationProvider.php [hunk 1]
```diff
@@ -71,7 +72,18 @@ public function getNavigationItems(array $options = [])
         $settings->setComponent('content/settings@sulucontent');
         $settings->setDisplay(['edit']);
 
-        $navigation = [$content, $seo, $excerpt, $settings];
+        $test = new ContentNavigationItem('test');
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ContentBundle/Admin/ContentContentNavigationProvider.php]: remove this after finishing this PR

**Comment by wachterjohannes** [comment]
@danrot @chirimoya before i implement this in all forms could one of you please review it shortly? i dont want to make things twice :)

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/vendor/husky/husky.js [hunk 2]
```diff
@@ -36320,8 +36345,51 @@ define('__component__$tabs@husky',[],function() {
             return Math.floor((Math.random() * 1677721500000000)).toString(16);
         },
 
+        /**
+         * Evaluates the given display-conditions against the condition-data.
+         *
+         * @param {[{property, operator, value}]} displayConditions
+         * @param {Object} conditionData
```
Comment (chirimoya) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/vendor/husky/husky.js]: `data`

**Comment by chirimoya** [comment]
else ACK

File: src/Sulu/Bundle/AdminBundle/Resources/public/js/vendor/husky/husky.js (hunk 2 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/vendor/husky/husky.js]: `data` is already used in the component. should i also rename that to `items`?

Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/vendor/husky/husky.js]: What exactly is `data` in this component?

Comment (wachterjohannes) [inline src/Sulu/Bundle/AdminBundle/Resources/public/js/vendor/husky/husky.js]: the "tab"-data

**Comment by wachterjohannes** [comment]
@danrot the issue with tabs is now also solved in this PR so this should be ready to review

File: src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/roles/main.js [hunk 3]
```diff
@@ -78,10 +76,14 @@ define([
         save: function(data, action) {
             this.sandbox.emit('sulu.header.toolbar.item.loading', 'save');
             this.role.set(data);
-            this.role.save(null, {
+
+            this.role.toJSON();
```
Comment (danrot) [inline src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/roles/main.js]: Does this call make any sense if the return value is not used?

File: src/Sulu/Bundle/SecurityBundle/Resources/public/js/models/permission.js [hunk 4]
```diff
@@ -1,22 +0,0 @@
...
-
-    return relationalModel({
-        defaults: function() {
-            return {
-                context: '',
-                permissions: ''
-            };
-        }
-    });
-});
```
Comment (danrot) [inline src/Sulu/Bundle/SecurityBundle/Resources/public/js/models/permission.js]: Why is this file deleted? Does it have to do anything with the tab conditions? Or was it simply not used?

File: src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/roles/main.js (hunk 3 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/roles/main.js]: i think this was for debugging i will remove it.

File: src/Sulu/Bundle/SecurityBundle/Resources/public/js/models/permission.js (hunk 4 above)
Comment (wachterjohannes) [inline src/Sulu/Bundle/SecurityBundle/Resources/public/js/models/permission.js]: it was never used exept in the role model. but this relation has no advantage because we return them with the same rest api endpoint

**Comment by wachterjohannes** [comment]
@danrot fixed and rebased
